api/controllers/v0.3/ILSHelper.js

```diff
@@ -0,0 +1,390 @@
+
+      // Simplistically assume person's name has two segments: the first
+      // and last name, with no middle names or honorifics.
+      // (e.g. "Dorothy Vaughn" "Katherine Johnson" "Mary Jackson")
+      let [firstName, lastName] = fullName.split(" ", 2);
+      firstName = firstName.trim();
+      lastName = lastName.trim();
+
+      if (firstName !== "" || lastName !== "") {
+        fullName = `//${lastName}, //${firstName}`;
+      } else {
+        // Our assumption of at least two names was wrong, and we got
+        // something like "Dorothy" or "Katherine " or " Jackson".
+        // We'll take what we can get.
+        fullName = firstName || lastName;
+      }
+    }
+
+    let fields = {
+      name: fullName,
+      address: address,
+      username: patron.username,
+      pin: patron.pin,
+      ptype: patron.ptype,
+      workAddress: workAddress,
+      ecommunications_pref: patron.ecommunicationsPref,
+    };
+
+    if (patron.barcode) {
+      fields["barcode"] = patron.barcode;
+    }
+    if (email) {
+      fields["email"] = email;
+    }
+    if (birthdate) {
+      fields["birthdate"] = birthdate;
+    }
+
+    return fields;
+  }
+
+  // Determine whether a patron should have a permanent card.
+  longExpiration(patron) {
+    // False if a patron's existing work address isn't commercial
+    if (patron.workAddress && patron.workAddress.isResidential) {
+      return false;
+    }
+
+    // False if patron provides a home address that is not residential
+    // False if patron does not have a recognized name
+    // False if patron policy is not the default (:simplye)
+    return (
+      patron.address.isResidential &&
+      patron.hasValidName &&
+      patron.policy.isDefault
+    );
+  }
+
+  patronFields(patron) {
+    let fieldData = this.formattedPatronData(patron);
+
+    let fields = [
+      fieldData["name"],
+      fieldData["address"],
+      fieldData["ptype"],
+      fieldData["username"],
+      fieldData["pin"],
+      fieldData["ecommunications_pref"],
+      // notice_pref_as_field,
+      // home_lib_as_field,
+      patron.policy.agency,
+    ];
+
+    if (patron.worksInCity) {
+      fields.push(fieldData["workAddress"]);
+      fields.push(note_as_field);
+    }
+
+    if (this.longExpiration(patron)) {
+      fields.push(this.longExpirationAsField(patron));
+    } else {
+      patron.set_temporary();
+      fields.push(this.shortExpirationAsField(patron));
+    }
+
+    // Add required/optional fields according to policy.
+    if (patron.barcode) {
+      fields.push(fieldData["barcode"]);
+    }
+    if (patron.email) {
+      fields.push(fieldData["email"]);
+    }
+    if (patron.birthdate) {
+      fields.push(fieldData["birthdate"]);
+    }
+
+    return {
+      patronFields: {
+        patron_field: fields,
+      },
+      patronID: patron.patronId,
+    };
+  }
+
+  blankField() {
+    return { fieldTag: "", marcTag: 0, value: "", value_is_binary: false };
```

Comment:
When would you want this?  Comment?

Reply:
Not sure, but just copied it over. This whole file will be updated soon.